src/h3lib/lib/h3Index.c

```diff
@@ -284,8 +285,9 @@ H3Index H3_EXPORT(cellToCenterChild)(H3Index h, int childRes) {
  * contiguous regions exist in the set at all and no compression possible)
  * @return an error code on bad input data
  */
+// todo: update internal implementation for int64_t
 int H3_EXPORT(compactCells)(const H3Index* h3Set, H3Index* compactedSet,
-                            const int numHexes) {
+                            const int64_t numHexes) {
```

Comment:
if you rebase you can also update the 4.x docs with this signature change & breaking change note. Not sure it's needed here since all the function signatures need to be updated.